In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import scipy.stats as stats
from scipy.stats import shapiro
import statsmodels.api as sm
from scipy.stats import ttest_rel

In [2]:
# Read case data
df = pd.read_excel("5. Case 4 - Paired samples t-test.xlsx")

In [3]:
# Print data head
df.head()

Customer ID  AOV_Before  AOV_After  AOV_Diff
0     43300313      114.17     126.08     11.91
1      8873290      132.21      90.30    -41.91
2     14352336       95.76     116.40     20.64
3     14864842       89.12     122.68     33.56
4     59734862      123.10     141.26     18.16

-	AOV_Before: Average Order Value of the last three customer purchases before receiving the email
-	AOV_After: Average Order Value of the last three customer purchases after receiving the email
-	AOV_Diff = AOV_After - AOV_Before: Difference in Customer Average Order Value -> dependent variable

In [4]:
# Descriptive statistics
df.describe()

Customer ID  AOV_Before   AOV_After    AOV_Diff
count  2.240000e+02  224.000000  224.000000  224.000000
mean   4.935132e+07  105.450848  110.814375    5.363527
std    2.854895e+07   21.981770   20.314000   31.911967
min    1.250455e+06   28.980000   42.010000  -81.650000
25%    2.590802e+07   91.785000   97.100000  -14.590000
50%    4.840174e+07  103.990000  110.595000    6.170000
75%    7.345052e+07  121.772500  126.167500   28.275000
max    9.994677e+07  156.440000  165.930000   92.650000

The table above reveals that the average order value after receiving the email (110.81) is higher than before (105.45). But is this difference statistically significant? This is what the paired sample t-test can help establish.

In [5]:
# Create arrays
data_diff = df["AOV_Diff"].values
data_before = df["AOV_Before"].values
data_after = df["AOV_After"].values

For the paired t-test, we require two variables: one to define the pairs of observations and the other as a measurement.

The null hypothesis is stated as follows:
- H0: µ(AOV_Before) - µ(AOV_After) = 0 ('the mean difference in AOV between the new and old return policy is zero').

We aim to demonstrate that the new return policy results in a higher AOV. Therefore, we adopt an upper-tailed alternative hypothesis (H1) indicating a positive difference between the two population means:
- H1: µ(AOV_Before) - µ(AOV_After) < 0 ('the mean difference in AOV between the new and old return policy is less than zero').

Furthermore, we set the significance level, alpha, to 0.05.

The **scipy.stats.ttest_rel()** function in SciPy is utilized to execute the paired samples t-test—also known as the dependent or related t-test. This test assesses whether there's a significant difference between the means of two paired or dependent samples.

The main parameters of the **ttest_rel function()** include:

1.	**a, b**: These arrays contain the sample data for comparison. Each should hold observations from one of the two related or paired samples, which must be of equal length and paired (e.g., pre- and post-treatment measurements on the same subjects or repeated measurements under different conditions).
2.	**alternative:** This parameter specifies the alternative hypothesis. Options include:
 -	**'two-sided'** checks whether the means of the underlying distributions of the samples are unequal (default). This is used to determine if there's any difference between the samples, regardless of direction.
 -	**'less'** checks if the mean of the first sample (a) is less than that of the second sample (b), implying a negative mean difference.
 -	**'greater'** checks if the mean of the first sample (a) is greater than that of the second sample (b), implying a positive mean difference. For our hypothesis, 'less' should be selected since we assert that the post-policy mean is greater (therefore, the mean difference is negative).

    
The degrees of freedom (df) are calculated as the number of pairs minus one.

In this scenario, df = 223.

In [6]:
# Run paired samples t-test
ttest_rel(data_after, data_before, alternative= "greater")

TtestResult(statistic=2.5154801271455183, pvalue=0.006295854693981037, df=223)

The p-value is smaller than the predefined alpha level of 0.05.

Consequently, we reject the null hypothesis in favor of the alternative. We conclude that the average order value under the new return policy (110.81 USD) is significantly higher than the old (105.45 USD) at the 0.05 significance level.